src/dropbox-base.js

```diff
@@ -265,10 +316,85 @@ export class DropboxBase {
             status: res.status,
           };
         }
+
+        if (data.refresh_token) {
+          return {
+            accessToken: data.access_token,
+            refreshToken: data.refresh_token,
+            accessTokenExpiresAt: getTokenExpiresAt(data.expires_in),
+          };
+        }
         return data.access_token;
       });
   }
 
+  /**
+   *
+   * @returns {Promise<[*, *]>}
+   */
+  checkAndRefreshAccessToken() {
+    const canRefresh = this.getRefreshToken() && this.getClientId() && this.getClientSecret();
+    const needsRefresh = this.getAccessTokenExpiresAt() &&
+            (new Date(Date.now() + TokenExpirationBuffer)) >= this.getAccessTokenExpiresAt();
+    const needsToken = !this.getAccessToken();
+    if ((needsRefresh || needsToken) && canRefresh) {
+      return this.refreshAccessToken();
+    }
+    return Promise.resolve();
+  }
+
+  /**
+   *
+   * @returns {Promise<[*, *]>}
+   */
+  refreshAccessToken() {
+    let refreshUrl = 'https://api.dropbox.com/oauth2/token';
```

Comment:
This feels like it should be defined as a constant already somewhere else? If not we should do that - I'd imagine it's also being used to construct the url for token exchange.

Reply:
Done in next revision